### DISCUSSION TITLE
PurgeCSS Removing MDX Styles in Nextjs Despite Being Included in Purge Content Option

### DISCUSSION THREAD

**Discussion by pdevito3** [tailwindlabs/tailwindcss#3017] [Help] [answered]
Hey all, so I have a Nextjs project that has some mdx pages for the docs section of the site. Everything is running fine locally, but when I deployed to vercel yesterday I noticed that my mdx pages were getting purged. I double checked that the content config included mdx files, but it's still wiping those styles. 

I was able to confirm that it was a purge issue by setting `enabled: true` in my purge setting, but I'm not sure how to resolve it. 

I even tried explicitly calling out the docs path in my purge settings. What am I missing?
```js
  purge: {
    content:[
      './src/**/*.{js,jsx,mdx}',
      './src/pages/docs/*.mdx',
    ]
  },
```

**Comment by pdevito3** [comment]
So in my myriad of code iterations, I gave this a go and it works... not sure why, but I'll take it. weird.

If anyone else is using this in the future, obviously make sure you take out the `enabled: true,` as that was just for testing

```js
  purge: {
    enabled: true,
    content:[
      './src/**/*.js',
      './*/*.{js,mdx}',
    ]
  },
```

**Reply by simonswiss** [reply]
I have unmarked this solution as the correct answer (it worked, but it wasn't the _solution_).

If you read the [new "marked answer"](https://github.com/tailwindlabs/tailwindcss/discussions/3017#discussioncomment-148508), you'll probably understand why this solution you landed on was also working 👍

**Comment by pdevito3** [comment]
@simonswiss any thoughts on why this is in case anyone else comes across this? Would be nice to know, but as long as it works, I guess I'm happy 😆

**Comment by simonswiss** [comment]
Hey!

Would it be possible to have a look at that repo? What version of Tailwind are you on? What does your Tailwind config look like?

**Reply by pdevito3** [reply]
Yup! TW 2.0. pulled several chunks of it from y'all website actually, so some it it might look familiar :-) 

https://github.com/pdevito3/wrapt.dev

**Comment by simonswiss** [comment, marked as answer]
Hola!

I had a look at your repo shared above, and I figured out what's happening.

The `prose` class name, which is responsible for applying all the styles from the Typography plugin, is nowhere to be found on any of the files within `./src/**/*.*`

That class is actually only present in plain text (what Purge CSS cares about) in the `withProse` file within the `remark` folder.

For that reason, purge strips out the `prose` class and kills all the Typography styles you rely on for the docs pages!

Your solution is to add the path to that `withProse` file in your `purge.content` array, or add `prose` as a `purge.options.safelist` value:

```diff
purge: {
  content: [
    "./src/**/*.{js,mdx}", 
+    "./remark/withProse.js"
  ],
},
```

or

```diff
purge: {
  content: ["./src/**/*.{js,mdx}"],
+  options: {
+    safelist: ["prose"],
+  },
},
```

Hope it helps you understand what was happening! 🎉

**Reply by pdevito3** [reply]
Damn that makes complete sense 🤦🏻‍♂️ Good find. Preciate ya!
